pyseq.py

```diff
@@ -879,3 +908,47 @@ def get_sequences(source):
     log.debug('time: %s' % (datetime.now() - start))
 
     return list(seqs)
+
+
+def walk(source, level=-1, topdown=True, onerror=None, followlinks=False):
+    """ Generator that traverses a directory structure starting at
+        source looking for sequences.
+        :param source: valid folder path to traverse
+        :param level: int, if < 0 traverse entire structure otherwise
+                      traverse to given depth
+        :param topdown: walk from the top down
+        :param onerror: callable to handle os.listdir errors
+        :param followlinks: whether to follow links
+    """
+    start = datetime.now()
+    assert isinstance(source, basestring) is True
+    assert os.path.exists(source) is True
+    source = os.path.abspath(source)
+    for root, dirs, files in os.walk(source, topdown, onerror, followlinks):
+        if topdown is True:
+            parts = root.replace(source, "").split(os.sep)
+            while "" in parts:
+                parts.remove("")
+            if len(parts) == level - 1:
+                del dirs[:]
+
+        files.sort(key=_natural_key)
+        seqs = []
```

Comment:
Is there a reason you didn't use `get_sequences()` here? I just noticed a bug where walk is not returning the correct sequences, e.g.
```
$ ./lss tests/files/ -r
...
├── fileA.0001.png
├── fileA.0002.jpg
├── fileA.0002.png
├── fileA.0003.jpg
├── fileA.0003.png
...
```
but using `seqs = get_sequences(files)` does return the correct sequences. 

I've already merged this PR and fixed this bug locally, so I'll go ahead and push the fix unless there's a reason not to do it this way.